webapp/packages/core-ui/src/Tabs/Tab/Tab.tsx

```diff
@@ -6,96 +6,65 @@
  * you may not use this file except in compliance with the License.
  */
 import { observer } from 'mobx-react-lite';
-import { useMemo, useRef, useState } from 'react';
+import { useMemo, useRef } from 'react';
 import { Tab as BaseTab } from 'reakit/Tab';
-import styled, { use } from 'reshadow';
 
-import { getComputed, Icon, useStyles, useTranslate } from '@cloudbeaver/core-blocks';
-import type { IDataContext } from '@cloudbeaver/core-data-context';
-import type { ComponentStyle } from '@cloudbeaver/core-theming';
+import { getComputed, s, useS, useTranslate } from '@cloudbeaver/core-blocks';
 import { useMenu } from '@cloudbeaver/core-view';
 
-import { ContextMenu } from '../../ContextMenu/ContextMenu';
 import { TabContext } from '../TabContext';
-import type { ITabsContext } from '../TabsContext';
-import { BASE_TAB_ACTION_STYLES, BASE_TAB_STYLES } from './BASE_TAB_STYLES';
 import { DATA_CONTEXT_TAB_ID } from './DATA_CONTEXT_TAB_ID';
 import { DATA_CONTEXT_TABS_CONTEXT } from './DATA_CONTEXT_TABS_CONTEXT';
 import { MENU_TAB } from './MENU_TAB';
+import style from './Tab.m.css';
+import { TabActions } from './TabActions';
 import type { TabProps } from './TabProps';
 import { useTab } from './useTab';
 
-export const Tab = observer<TabProps>(function Tab({ tabId, title, menuContext, disabled, className, children, style, onOpen, onClose, onClick }) {
+export const Tab = observer<TabProps>(function Tab(props) {
   const translate = useTranslate();
+  const { tabId, title, disabled, className, children, onOpen, onClose, onClick } = props;
   const ref = useRef<HTMLButtonElement>(null);
   const tabContext = useMemo(() => ({ tabId }), [tabId]);
   const tab = useTab(tabId, onOpen, onClose, onClick);
   const info = tab.getInfo();
-
+  const styles = useS(style);
   const canClose = getComputed(() => !!onClose || (tab.closable && tab.state.closable));
 
-  return styled(useStyles(BASE_TAB_STYLES, BASE_TAB_ACTION_STYLES, style))(
+  const menu = useMenu({
+    menu: MENU_TAB,
+    context: props.menuContext,
+  });
+
+  menu.context.set(DATA_CONTEXT_TABS_CONTEXT, tab.state);
+  menu.context.set(DATA_CONTEXT_TAB_ID, tabId);
```

Comment:
please move it to TabMenu component and use this component in TabActions or better to use composition and pass this menuItem as children:
```tsx
<TabActions>
  {canClose && <TabAction name="cross-bold" title={translate('ui_close')} onClick={tab.handleClose} />}
    <TabMenu tabId={tabId} state={tab.state} menuContext={menuContext} />
</TabActions>

Reply:
as i can see you probably left this code accedently (it's duplicated in TabMenu)

Reply:
done